BB-362 remove kafka connect connector config from logs

welansari opened the pull request:
kafka connect connector config contains the mongodb connection string which should not be logged

also fixing a bug with the oplog populator where if a connector update fails cause the connector can't be found, it is not considered an error

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/backbeat/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-362 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.6.5`


Please check the `Fix Version/s` of BB-362, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-362 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.4.26`

* `8.5.5`

* `8.6.5`


Please check the `Fix Version/s` of BB-362, or the target
branch of this pull request.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/BB-362-fix-oplog-populator-mongo-creds-log` into
`development/8.4`
* [w/8.5/bugfix/BB-362-fix-oplog-populator-mongo-creds-log](https://github.com/scality/backbeat/compare/development/8.5...w/8.5/bugfix/BB-362-fix-oplog-populator-mongo-creds-log) will be merged into `development/8.5` 
* [w/8.6/bugfix/BB-362-fix-oplog-populator-mongo-creds-log](https://github.com/scality/backbeat/compare/development/8.6...w/8.6/bugfix/BB-362-fix-oplog-populator-mongo-creds-log) will be merged into `development/8.6` 



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/8.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

bert-e commented:
# History mismatch



Merge commit #9008c54bcceac04af74a3b009c3f7b2c01c72900 on the integration branch
`w/8.5/bugfix/BB-362-fix-oplog-populator-mongo-creds-log` is merging a branch which is neither the current
branch `bugfix/BB-362-fix-oplog-populator-mongo-creds-log` nor the development branch
`development/8.5`.

It is likely due to a rebase of the branch `bugfix/BB-362-fix-oplog-populator-mongo-creds-log` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

Kerkesni commented:
/reset

bert-e commented:
# Lossy reset warning



There seems to be *manual* commits on integration branches (e.g. conflict
resolutions) that will be lost if you chose to *reset*.

**You can use the `force_reset` command if you still want me
to delete those branches.**

bert-e commented:
# History mismatch



Merge commit #9008c54bcceac04af74a3b009c3f7b2c01c72900 on the integration branch
`w/8.5/bugfix/BB-362-fix-oplog-populator-mongo-creds-log` is merging a branch which is neither the current
branch `bugfix/BB-362-fix-oplog-populator-mongo-creds-log` nor the development branch
`development/8.5`.

It is likely due to a rebase of the branch `bugfix/BB-362-fix-oplog-populator-mongo-creds-log` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

Kerkesni commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/BB-362-fix-oplog-populator-mongo-creds-log` into
`development/8.4`
* [w/8.5/bugfix/BB-362-fix-oplog-populator-mongo-creds-log](https://github.com/scality/backbeat/compare/development/8.5...w/8.5/bugfix/BB-362-fix-oplog-populator-mongo-creds-log) will be merged into `development/8.5` 
* [w/8.6/bugfix/BB-362-fix-oplog-populator-mongo-creds-log](https://github.com/scality/backbeat/compare/development/8.6...w/8.6/bugfix/BB-362-fix-oplog-populator-mongo-creds-log) will be merged into `development/8.6` 



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/8.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand commented on lib/wrappers/KafkaConnectWrapper.js:
```diff
@@ -242,18 +241,16 @@ class KafkaConnectWrapper {
             connector: connectorName,
         });
-        return undefined;
+        throw errors.InternalError.customizeDescription('Could not find connector to update');
```
not the same commit : this is unrelated to the logs...

francoisferrand reviewed (commented):
.

bert-e commented:
# History mismatch



Merge commit #f7575acc8fe98878561f19036fdf1b40036945a2 on the integration branch
`w/8.5/bugfix/BB-362-fix-oplog-populator-mongo-creds-log` is merging a branch which is neither the current
branch `bugfix/BB-362-fix-oplog-populator-mongo-creds-log` nor the development branch
`development/8.5`.

It is likely due to a rebase of the branch `bugfix/BB-362-fix-oplog-populator-mongo-creds-log` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

Kerkesni commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/BB-362-fix-oplog-populator-mongo-creds-log` into
`development/8.4`
* [w/8.5/bugfix/BB-362-fix-oplog-populator-mongo-creds-log](https://github.com/scality/backbeat/compare/development/8.5...w/8.5/bugfix/BB-362-fix-oplog-populator-mongo-creds-log) will be merged into `development/8.5` 
* [w/8.6/bugfix/BB-362-fix-oplog-populator-mongo-creds-log](https://github.com/scality/backbeat/compare/development/8.6...w/8.6/bugfix/BB-362-fix-oplog-populator-mongo-creds-log) will be merged into `development/8.6` 



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/8.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

Commit:
BB-362 remove kafka connect connector config from logs

kafka connect connector config contains the mongodb connection string
which should not be logged

Commit:
BB-362 fix oplog populator not retrying to update connector when connector is not found

fixing a bug with the oplog populator where if a connector update
fails cause the connector can't be found, it is not considered an error
and the update is not retried later

bert-e commented:
# History mismatch



Merge commit #ffe55021cc8582770d904e030b450e8022a5ff2e on the integration branch
`w/8.5/bugfix/BB-362-fix-oplog-populator-mongo-creds-log` is merging a branch which is neither the current
branch `bugfix/BB-362-fix-oplog-populator-mongo-creds-log` nor the development branch
`development/8.5`.

It is likely due to a rebase of the branch `bugfix/BB-362-fix-oplog-populator-mongo-creds-log` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

welansari commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/BB-362-fix-oplog-populator-mongo-creds-log` into
`development/8.4`
* [w/8.5/bugfix/BB-362-fix-oplog-populator-mongo-creds-log](https://github.com/scality/backbeat/compare/development/8.5...w/8.5/bugfix/BB-362-fix-oplog-populator-mongo-creds-log) will be merged into `development/8.5` 
* [w/8.6/bugfix/BB-362-fix-oplog-populator-mongo-creds-log](https://github.com/scality/backbeat/compare/development/8.6...w/8.6/bugfix/BB-362-fix-oplog-populator-mongo-creds-log) will be merged into `development/8.6` 



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/8.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

welansari commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.4`

* :heavy_check_mark: `development/8.5`

* :heavy_check_mark: `development/8.6`



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/8.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.4`

* :heavy_check_mark: `development/8.5`

* :heavy_check_mark: `development/8.6`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/8.3`



Please check the status of the associated issue BB-362.

Goodbye kerkesni.
